packages/flutter_tools/lib/src/version.dart

```diff
@@ -207,20 +207,34 @@ class FlutterVersion {
   /// The amount of time we wait before pinging the server to check for the
   /// availability of a newer version of Flutter.
   @visibleForTesting
-  static const Duration kCheckAgeConsideredUpToDate = Duration(days: 3);
+  static const Duration checkAgeConsideredUpToDate = Duration(days: 3);
 
   /// We warn the user if the age of their Flutter installation is greater than
-  /// this duration.
+  /// this duration. The durations are slightly longer than the expected release
+  /// cadence for each channel, to give the user a grace period before they get
+  /// notified.
   ///
-  /// This is set to 5 weeks because releases are currently around every 4 weeks.
+  /// For example, for the beta channel, this is set to five weeks because
+  /// beta releases happen approximately every month.
   @visibleForTesting
-  static const Duration kVersionAgeConsideredUpToDate = Duration(days: 35);
+  static Duration versionAgeConsideredUpToDate(String channel) {
+    switch (channel) {
+      case 'stable':
+        return const Duration(days: 7 * 4 * 4);
```

Comment:
i recommend making this at least 6 months, maybe 12

Reply:
Done